### PR TITLE
fixing crash related to overflowing the LRU cache

### PR DESCRIPTION
**Pull request by isaacrob-roboflow**
# Description

OWLv2 implementation crashed if it received more than 50 prompt images. I fixed this by making sure we use the image's embedding as soon as it is computed, instead of allowing it to sit in a cache that may have to reject it due to too many objects. I also increased the cache size so that shouldn't be a problem in the first place, but at least now it is CERTAIN that there won't be a crash related to that issue.

I also improved the integration test to include two positive prompts and one negative prompt.

## Type of change

Please delete options that are not relevant.

-   [x] Bug fix (non-breaking change which fixes an issue)
-   [ ] New feature (non-breaking change which adds functionality)
-   [ ] This change requires a documentation update

## How has this change been tested, please provide a testcase or example of how you tested the change?

Tested via the integration test, and by running a sample with more (unique) images than the cache size and verifying that it didn't break.

## Any specific deployment considerations

For example, documentation changes, usability, usage/costs, secrets, etc.

## Docs

-   [ ] Docs updated? What were the changes:

### CODE REVIEW

**Review by probicheaux — APPROVED** [review]

**Comment by probicheaux** [comment]
ah @PawelPeczek-Roboflow can you review

File: docker/dockerfiles/Dockerfile.onnx.gpu [hunk 1]
```diff
@@ -60,7 +60,7 @@ WORKDIR /build
 COPY . .
 RUN ln -s /usr/bin/python3 /usr/bin/python
 RUN /bin/make create_wheels_for_gpu_notebook
-RUN pip3 install dist/inference_cli*.whl dist/inference_core*.whl dist/inference_gpu*.whl dist/inference_sdk*.whl
+RUN pip3 install --no-cache-dir dist/inference_cli*.whl dist/inference_core*.whl dist/inference_gpu*.whl dist/inference_sdk*.whl
```
Comment (grzegorz-roboflow) [inline docker/dockerfiles/Dockerfile.onnx.gpu:63]: Should we be consistent and do the same in `Dockerfile.onnx.cpu.dev`, `Dockerfile.onnx.gpu` and `Dockerfile.onnx.gpu.parallel`?

File: docker/dockerfiles/Dockerfile.onnx.gpu.dev [hunk 2]
```diff
@@ -58,9 +58,6 @@ RUN python3 -m pip install packaging==24.1  && rm -rf ~/.cache/pip
 # Install flash_attn required for Paligemma and Florence2
 RUN python3 -m pip install -r requirements.pali.flash_attn.txt --no-build-isolation && rm -rf ~/.cache/pip
 
-FROM scratch
-COPY --from=base / /
-
```
Comment (grzegorz-roboflow) [inline docker/dockerfiles/Dockerfile.onnx.gpu.dev:63]: There are other docker files where these were not removed, do we want to have only this single image without COPY all layer?

**Review by PawelPeczek-Roboflow — APPROVED** [review]
